### PR TITLE
getcmdline_prompt: Temporarily disable msg_silent so prompt is displayed

### LINKED ISSUE #7378: :silent input() is too silent
As discussed [on gitter](https://gitter.im/neovim/neovim?at=59de16fe3cb340a2610fa8e8)

In neovim `:silent let foo = input('Foo: ')` doesn't prompt, nor does it echo the input. This is different from the vim behavior.

- `nvim --version`:
```
NVIM v0.2.1-939-gdb999821d
Build type: RelWithDebInfo
Compilation: /usr/bin/gcc-5 -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/build/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@testing-gce-2baa95ab-096b-4da3-8e16-eda660b6445d

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/home/travis/build/neovim/bot-ci/build/neovim/build/nvim.AppDir/usr/share/nvim
"
```
- Vim (version: 8.0.1184) behaves differently? yes
- Operating system/version: Fedora 26
- Terminal name/version: GNOME Terminal 3.26.0 Using VTE version 0.50.1 +GNUTLS
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`
```
nvim -u NORC
:silent let foo = input('Foo: ')
```

### Actual behaviour
no prompt, and no echo of input (though the cursor advances)

### Expected behaviour
prompt and echo input

### PR DESCRIPTION
**Pull request by jamessan**
Closes #7378

### CODE REVIEW

**Comment by justinmk** [comment]
I think@zyx-i worked on this, maybe he should approve .

**Comment by ZyX-I** [comment]
I am not sure about why it stopped working in first place, would need some time to investigate (at least on the level “it stopped working (not) after my PR/(not) after global refactoring of input handling that took place earlier”). But in any case test is missing.

**Comment by jamessan** [comment]
> I am not sure about why it stopped working in first place

It didn't stop working.  Vim fixed it in 7.4.1636, which we mistakenly marked as NA.

> But in any case test is missing.

I'll fix that today.

**Comment by jamessan** [comment]
> > But in any case test is missing.
> 
> I'll fix that today.

Done.

File: src/nvim/ex_getln.c [hunk 1]
```diff
@@ -1863,7 +1863,10 @@ char *getcmdline_prompt(const char firstc, const char *const prompt,
   ccline.input_fn = (firstc == '@');
   ccline.highlight_callback = highlight_callback;
 
+  int msg_silent_saved = msg_silent;
+  msg_silent = 0;
   char *const ret = (char *)getcmdline(firstc, 1L, 0);
+  msg_silent = msg_silent_saved;
 
   restore_cmdline(&save_ccline);
```
Comment (justinmk) [inline src/nvim/ex_getln.c]: in the vim patch, this comes before the restore of msg_silent. Doesn't matter, but  good to keep it consistent unless there's a reason not to.

**Comment by justinmk** [comment]
LGTM. Perhaps also mention 7.4.1636 in the commit message.

**Pull request merged**